core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -263,13 +263,18 @@ coreos:
       command: start
       content: |
         [Unit]
-        Requires={{.Etcd.SystemdUnitName}}
+        Wants={{.Etcd.SystemdUnitName}}
         After={{.Etcd.SystemdUnitName}}
 
         [Service]
-        Type=oneshot
+        Type=simple
+        Restart=on-failure
+        RestartSec=10
+
         EnvironmentFile={{.EtcdNodeEnvFileName}}
-        ExecStart=/opt/bin/cfn-signal
+        ExecStartPre=/usr/bin/systemctl is-active {{.Etcd.SystemdUnitName}}
+        ExecStartPre=/usr/bin/rkt fetch --pull-policy new {{.AWSCliImage.Options}}{{.AWSCliImage.RktRepo}}
+        ExecStart=-/opt/bin/cfn-signal
```

Comment:
@redbaron Thanks!
Yes, that is one of cases. One more case is that, as cfn-signal is just an http request to a wait handle url under the hood, it can fail temporarily.
Ideally, cfn-signal should be skipped or allowed to fail without retyring only when CF doesn't even expect the signal.
However, the only way I can think of is to grep the error message from cfn-signal and script /opt/bin/cfn-signal to act accordingly. Not elegant :)